### PR TITLE
Fix kubeconfig priority

### PR DESCRIPTION
**Pull request by LeoLiuYan**

### CODE REVIEW

**Comment by LeoLiuYan** [comment]
ping @yasker

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@
 /dist
 *.swp
 .idea
+.vscode/
```
Comment (yasker) [inline .gitignore:7]: Can you separate this into another commit?

File: main.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ import (
 )
 
 var (
-	VERSION = "0.0.1"
+	VERSION = "0.0.17"
```
Comment (yasker) [inline main.go:22]: Don't need to change the VERSION variable manually since it should be changed while compiling the binary. You can run `./bin/local-path-provisioner --version` to verify.

File: main.go [hunk 3]
```diff
@@ -110,17 +110,25 @@ func homeDir() string {
...
+			if _, err := os.Stat(f); err == nil {
+				return clientcmd.BuildConfigFromFlags("", f)
+			}
+		}
 	}
-	_, err := os.Stat(kubeconfig)
-	if err != nil {
-		return nil, err
+
+	if c, err := rest.InClusterConfig(); err == nil {
```
Comment (yasker) [inline main.go:127]: Hmm? So the priority would be:
1. ENV/RecommendedDir
2. In-cluster
3. RecommendedFileName 

Is this intended? I found it's weird that in-cluster is behind recommended dir but before the recommended file.

Also, do we still need `DefaultKubeConfigFilePath`?

Comment (LeoLiuYan) [reply main.go:126]: Oh... The priority of kubeconfig would be:

1.  Command args.  We could specify the --kubeconfig.
2. ENV.  We read from RecommendedConfigPathEnvVar.
3. In-cluster.
4. Recommended. We got by RecommendedHomeDir and RecommendedFileName.

Comment (yasker) [reply main.go:126]: That makes sense. Can you add this order to the commit message? Otherwise LGTM.

Comment (LeoLiuYan) [reply main.go:126]: OK.
